xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/test/java/org/xwiki/web/MacrosTest.java

```diff
@@ -287,4 +297,78 @@ void displayUser() throws Exception
             + "<a class=\"user-name\" href=\"/xwiki/bin/view/XWiki/Foo\">Foo</a>"
             + "</div>", out.toString().trim());
     }
+
+    @Test
+    void getSanitizedURLAttributeValue() throws Exception
+    {
+        HTMLScriptService htmlScriptService = mock(HTMLScriptService.class);
+        this.oldcore.getMocker().registerComponent(ScriptService.class, "html", htmlScriptService);
+        SecurityScriptService securityScriptService = mock(SecurityScriptService.class);
+        this.oldcore.getMocker().registerComponent(ScriptService.class, "security", securityScriptService);
+        URLSecurityScriptService urlSecurityScriptService = mock(URLSecurityScriptService.class);
+        this.oldcore.getMocker().registerComponent(ScriptService.class, "security.url", urlSecurityScriptService);
+        when(securityScriptService.get("url")).thenReturn(urlSecurityScriptService);
```

Comment:
you probably don't event need to register "security.url" as it is revolved by `thenReturn` instead of a component manager inside the `get`

Reply:
right